docs/secure-communication-agents.asciidoc

```diff
@@ -0,0 +1,89 @@
+[[secure-communication-agents]]
+== Secure communication with APM Agents
+
+To secure the communication between APM Agents and the APM Server:
+
+. <<secret-token,Set a secret token in your Agents and Server>>
+. <<ssl-setup,Enable SSL/TLS in APM Server>>
+. <<https-in-agents,Enable HTTPS in your APM Agents>>
+
+[[secret-token]]
+[float]
+=== Secret token
+
+You can configure a secret token to authorize requests to the APM Server.
+This ensures that only your agents are able to send data to your APM servers.
+Both the agents and the APM servers have to be configured with the same secret token,
+and secret tokens only provide security when used in combination with SSL/TLS.
+
+**APM Server configuration**
+
+Here's how you set the secret token in APM Server:
+
+[source,yaml]
+----
+apm-server.secret_token: <secret-token>
+----
+
+We recommend saving the token in the APM Server <<keystore>>.
+
+IMPORTANT: Secret tokens are not applicable for the RUM Agent,
+as there is no way to prevent them from being publicly exposed.
+
+**Agent specific configuration**
+
+Each Agent has a configuration for setting the value of the secret token:
+
+* *Go Agent*: {apm-go-ref}/configuration.html#config-secret-token[`ELASTIC_APM_SECRET_TOKEN`]
+* *Java Agent*: {apm-java-ref}/config-reporter.html#config-secret-token[`secret_token`]
+* *.NET Agent*: {apm-dotnet-ref}/config-reporter.html#config-secret-token[`ELASTIC_APM_SECRET_TOKEN`]
+* *Node.js Agent*: {apm-node-ref}/configuration.html#secret-token[`Secret Token`]
+* *Python Agent*: {apm-py-ref}/configuration.html#config-secret-token[`secret_token`]
+* *Ruby Agent*: {apm-ruby-ref}/configuration.html#config-secret-token[`secret_token`]
+
+[[ssl-setup]]
+[float]
+=== SSL/TLS communication in APM Server
+
+To enable SSL/TLS, you need to enable SSL and provide both a private key and a certificate
+issued by a certification authority (CA).
```

Comment:
We use "certificate authority" in the other existing pages (as well as below on this page!), should probably change this to be consistent? I think "certificate authority" is more typical anyway.